File: csharp/Range.cs
using System;
using System.Collections.Generic;

namespace fizzbuzz {
    interface IRange {
        IValue[] Values();
    }

    class IntRange : IRange {
        private int len;

        public IntRange(int len) {
            this.len = len;
        }

        public IValue[] Values() {
            var values = new IntValue[len];
            for(int count = 1; count <= len; count++) {
                values[count - 1] = new IntValue(count);
            }
            return values;
        }
    }
}

File: haskell/src/Range.hs
module Range(IRange(..), intRange) where

import Value(IValue(..), intValue)

data IRange a = IRange
    { values    :: [IValue a] }

createList :: [IValue a] -> (IValue a) -> (IValue a) -> [IValue a]
createList currLs currVal endVal
    | ((equal currVal) currVal endVal) = compoundLs
    | otherwise = createList compoundLs nextVal endVal
    where
        compoundLs = currLs ++ [ currVal ]
        nextVal = ((next currVal) currVal)

intRange :: Int -> (IRange Int)
intRange len =
    IRange  { values =  createList [] (intValue 1) (intValue len) }
